design-proposals/cluster-orch-k3s.md

```diff
@@ -0,0 +1,121 @@
+# Design Proposal: Add suppport for K3s
+
+Author(s): Hyunsun Moon, Andy Bavier, Denisio Togashi
+
+Last updated: 4/28/25
+
+## Abstract
+
+This proposal aims to enhance the Cluster Orchestration by introducing K3s as an additional option for edge cluster deployment. Currently, the framework supports only RKE2. By incorporating K3s, a production-grade, CNCF-certified lightweight and simplified Kubernetes distribution, the framework will allow users to select the most appropriate Kubernetes distribution for their specific use cases and the resource constraints of their environments. K3s will be set as the default choice, as it meets the needs of most deployment scenarios with its efficient resource usage, while RKE2 will remain available for users requiring a full Kubernetes experience for more complex applications. This change not only enhances flexibility but also lays the groundwork for potentially integrating additional Kubernetes distributions in the future, further expanding the framework's versatility and adaptability to diverse deployment scenarios.
```

Comment:
do we have examples where RKE2 is needed and the only option for an user ?

Reply:
Not that I know of